src/common.js

```diff
@@ -53,10 +57,12 @@ ext.resource._register();
  * DBus Interface Introspection
  */
 ext.dbusinfo = new Gio.DBusNodeInfo.new_for_xml(
-    Gio.resources_lookup_data(
-        ext.app_path + "/" + ext.app_id + ".xml",
-        0
-    ).toArray().toString()
+    ByteArray.toString(
+        Gio.resources_lookup_data(
+             ext.app_path + "/" + ext.app_id + ".xml",
+             0
+        ).toArray()
+    )
 );
```

Comment:
You could just include a function since it's used a few places:

```js
gsconnect.get_resource = function(rel_path) {
    let array = Gio.resources_lookup_data(
        GLib.build_filenamev([gsconnect.app_path, rel_path]),
        Gio.ResourceLookupFlags.NONE
    ).toArray();

    if (array instanceof Uint8Array) {
        array = imports.byteArray.toString(data);
    } else {
        array = array.toString();
    }

    return array.replace('@DATADIR@', ext.datadir);
};
```

Reply:
That was easy enough with a if/else, good to know. And good thinking with the function.
Want me to cleanup the PR or scrap it? Either way works for me. Since you're confident in v13 I dont feel huge need to rely on legacy branch anyway. :)

Reply:
it's up to you, they're just warnings anyways and I think anyone with 3.28+ will end up updating before it's completely deprecated.

Reply:
True, will scrap it since I've moved on to master/v13 regardless :)